src/rust/cryptography-x509-validation/src/policy/mod.rs

```diff
@@ -119,28 +178,103 @@ pub struct Policy<'a, B: CryptoOps> {
     /// The validation time. All certificates validated by this policy must
     /// be valid at this time.
     pub validation_time: asn1::DateTime,
+
+    /// An extended key usage that must appear in EEs validated by this policy.
+    pub extended_key_usage: ObjectIdentifier,
+
+    /// The set of permitted signature algorithms, identified by their
+    /// algorithm identifiers.
+    ///
+    /// If not `None`, all certificates validated by this policy MUST
+    /// have a signature algorithm in this set.
+    ///
+    /// If `None`, all signature algorithms are permitted.
+    pub permitted_algorithms: Option<HashSet<AlgorithmIdentifier<'a>>>,
+
+    pub(crate) critical_ca_extensions: HashSet<ObjectIdentifier>,
+    pub(crate) critical_ee_extensions: HashSet<ObjectIdentifier>,
```

Comment:
Do I understand right that the point of these to enforce that some extensions must be marked critical?

@reaperhulk let me know if you disagree, but I don't think we care about enforcing that: it's a lot of compatibility cost, for effectively no gain. I don't believe any other validators enforce "this extension must be marked critical", just the inverse (rejecting unknown critical extensions).